qt/widgets/instrumentview/src/PanelsSurface.cpp

```diff
@@ -519,24 +492,60 @@ std::pair<std::vector<size_t>, Mantid::Kernel::V3D> PanelsSurface::processUnstru
     }
     detectors.emplace_back(child);
   }
-  return std::make_pair(detectors, normal);
+  if (detectors.size() > 1) {
+    int index = m_flatBanks.size();
+    // save bank info
+    auto *info = new FlatBankInfo(this);
```

Comment:
My C++ is a little rusty, but shouldn't there be a `delete` call for this object (maybe there is and I've missed it!)?

Reply:
This piece of code was moved or copied by me in this PR rather than me having created it myself so hadn't studied this question in detail. Yes there should be a delete. The info variable gets added to a list `m_flatBanks` and then delete gets called on all the entries in here in `PanelsSurface::clearBanks` which is called from the `PanelsSurface` destructor. So think it's all OK here